lib/compiler/passes/generate-js.js

```diff
@@ -473,7 +473,7 @@ function generateJS(ast, options) {
 
           if (thenSp !== elseSp) {
             throw new Error(
-              "Branches of a condition must move the stack pointer in the same way."
+              "Branches of a condition must move the stack pointer in the same way (then=" + thenSp + ", else=" + elseSp + ")."
```

Comment:
I’d format the message like this for more readability: “Branches of a condition must move the stack pointer in the same way (before: 5, after then: 6, after else: 4).”. Or maybe even like this: “Branches of a condition can't move the stack pointer differently (before: 5, after then: 6, after else: 4).”.

Also please split the concatenation over multiple lines.